js/bitstamp.js

```diff
@@ -1152,20 +1452,28 @@ module.exports = class bitstamp extends Exchange {
     }
 
     async withdraw (code, amount, address, tag = undefined, params = {}) {
+        // For fiat withdrawals please provide all required additional parameters in the 'params'
+        // Check https://www.bitstamp.net/api/ under 'Open bank withdrawal' for list and description.
         this.checkAddress (address);
-        if (this.isFiat (code)) {
-            throw new NotSupported (this.id + ' fiat withdraw() for ' + code + ' is not implemented yet');
-        }
-        const name = this.getCurrencyName (code);
         const request = {
             'amount': amount,
-            'address': address,
         };
-        const v1 = (code === 'BTC');
-        let method = v1 ? 'v1' : 'private'; // v1 or v2
-        method += 'Post' + this.capitalize (name) + 'Withdrawal';
-        if (tag !== undefined) {
-            request['destination_tag'] = tag;
+        let method = undefined;
+        if (!this.isFiat (code)) {
+            const name = this.getCurrencyName (code);
+            const v1 = (code === 'BTC');
+            method = v1 ? 'v1' : 'private'; // v1 or v2
+            method += 'Post' + this.capitalize (name) + 'Withdrawal';
+            if (code === 'XRP') {
+                if (tag !== undefined) {
+                    request['destination_tag'] = tag;
+                }
+            }
+            request['address'] = address;
+        } else {
+            method = 'privatePostWithdrawalOpen';
+            request['iban'] = address;
+            request['account_currency'] = code;
```

Comment:
We should send the exchange-specific `currency['id']`, not the unified `code` here, and we have to add the `await this.loadMarkets()` call to `withdraw()` and `fetchDepositAddress()`, since those methods refer to markets and/or currencies:

```JavaScript
    async withdraw (code, amount, address, tag = undefined, params = {}) {
        ...
        request['account_currency'] = code;
```

↓

```JavaScript
    async withdraw (code, amount, address, tag = undefined, params = {}) {
        await this.loadMarkets();
        ...
        const currency = this.currency (code);
        request['account_currency'] = code;
```

I would add these edits, however, I can't edit the PR files. Can you please commit these modifications?

Reply:
Sorry, totally missed that one. Please check the committed changes.

Reply:
@domenus thank you so much for your involvement!